galaxy_ng/app/dynaconf_hooks.py

```diff
@@ -709,8 +712,37 @@ def read_settings_from_cache_or_db(
 
         return temp_settings.get(key, value.value)
 
+    def alter_hostname_settings(
+        temp_settings: Settings,
+        value: HookValue,
+        key: str,
+        *args,
+        **kwargs
+    ) -> Any:
+        """Use the request headers to dynamically alter the content origin and api hostname.
+        This is useful in scenarios where the hub is accessible directly and through a
+        reverse proxy.
+        """
+
+        ALLOWED_KEYS = ['CONTENT_ORIGIN', 'ANSIBLE_API_HOSTNAME']
+
+        if not apps.ready or key.upper() not in DYNAMIC_SETTINGS_SCHEMA:
+            # If app is starting up or key is not on allowed list bypass and just return the value
+            return value.value
+
+        if key.upper() in ALLOWED_KEYS:
+            req = get_current_request()
+            if req is not None:
+                headers = dict(req.headers)
+                proto = headers.get('X-Forwarded-Proto', 'http')
+                host = headers.get('Host', 'localhost:5001')
+                baseurl = proto + '://' + host
+                return baseurl
+
+        return temp_settings.get(key, value.value)
+
     return {
         "_registered_hooks": {
-            Action.AFTER_GET: [Hook(read_settings_from_cache_or_db)]
+            Action.AFTER_GET: [Hook(read_settings_from_cache_or_db), Hook(alter_hostname_settings)]
```

Comment:
We can probably change these lines, and have the hooks to be registered conditionally, instead of using the `GALAXY_DYNAMIC_SETTINGS` check above.

Something like:

On settings:

```py
DYNACONF_AFTER_GET_HOOKS=["read_settings_from_cache_or_db", "alter_hostname_settings"]
```

Here:

```python
# replace the check for dynamic_settings above
enabled_hooks = settings.get("DYNACONF_AFTER_GET_HOOKS")
if not enabled_hooks:
    return

...

return {
   "_registered_hooks": {
       Action.AFTER_GET: [Hook(locals()[function]) for function in enabled_hooks]
   }
}
```

the use of `locals` require that hook function is always defined inside the dynaconf_hooks file.